Operator Has a Restart in GCP Cluster

### Environment info
- NooBaa Operator Version: 5.14
- Platform: Kubernetes v1.27.8+4fab27b | OpenShift 4.14.0-0.nightly-2023-12-01-194244

### Actual behavior
1. When installing noobaa on GCP Cluster we see that the operator pod has 1 resart.

### Expected behavior
1. No restart of the operator pod after noobaa install. 

### Steps to reproduce
1. Use the cluster-bot with `launch 4.14 gcp`.
2. Install noobaa with latest image: `noobaa install --dev --noobaa-image='noobaa/noobaa-core:master-20231202'  --operator-image='noobaa/noobaa-operator:master-20231202' -n test1`

### More information - Screenshots / Logs / Other output
Attached logs
[operatolr_logs_gcp_previous.txt](https://github.com/noobaa/noobaa-operator/files/13538891/operatolr_logs_gcp_previous.txt)
[operatolr_logs_gcp.txt](https://github.com/noobaa/noobaa-operator/files/13538893/operatolr_logs_gcp.txt)

This is the relevant part:
```
time="2023-12-03T11:47:17Z" level=info msg="❌ Not Found: BackingStore \"noobaa-default-backing-store\"\n"
time="2023-12-03T11:47:17Z" level=info msg="CredentialsRequest \"noobaa-gcp-cloud-creds\" created.  creating default backing store on GCP objectstore" func=ReconcileDefaultBackingStore sys=test1/noobaa
time="2023-12-03T11:47:17Z" level=info msg="✅ Exists:  \"noobaa-gcp-cloud-creds-secret\"\n"
time="2023-12-03T11:47:17Z" level=info msg="Secret noobaa-gcp-cloud-creds-secret was created successfully by cloud-credentials operator" sys=test1/noobaa
time="2023-12-03T11:47:17Z" level=info msg="❌ Not Found: Secret \"noobaa-gcp-bucket-creds\"\n"
time="2023-12-03T11:47:17Z" level=info msg="Create event detected for noobaa-gcp-bucket-creds (test1), queuing Reconcile"
time="2023-12-03T11:47:17Z" level=info msg="checking which backingstore to reconcile. mapping secret test1/noobaa-gcp-bucket-creds to backingstores"
time="2023-12-03T11:47:17Z" level=info msg="Create event detected for noobaa-gcp-bucket-creds (test1), queuing Reconcile"
time="2023-12-03T11:47:17Z" level=info msg="checking which namespaceStores to reconcile. mapping secret test1/noobaa-gcp-bucket-creds to namespaceStores"
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 708 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	noobaa-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:115 +0x1fa
panic({0x2365900, 0x2cf4b40})
	/opt/hostedtoolcache/go/1.20.11/x64/src/runtime/panic.go:884 +0x213
github.com/noobaa/noobaa-operator/v5/pkg/system.(*Reconciler).prepareGCPBackingStore(0xc00045aa00)
	noobaa-operator/pkg/system/phase4_configuring.go:959 +0x536
github.com/noobaa/noobaa-operator/v5/pkg/system.(*Reconciler).ReconcileDefaultBackingStore(0xc00045aa00)
	noobaa-operator/pkg/system/phase4_configuring.go:716 +0x35a
github.com/noobaa/noobaa-operator/v5/pkg/system.(*Reconciler).ReconcilePhaseConfiguring(0xc00045aa00)
	noobaa-operator/pkg/system/phase4_configuring.go:76 +0x12e
github.com/noobaa/noobaa-operator/v5/pkg/system.(*Reconciler).ReconcilePhases(0x27afb57?)
	noobaa-operator/pkg/system/reconciler.go:547 +0x65
github.com/noobaa/noobaa-operator/v5/pkg/system.(*Reconciler).Reconcile(0xc00045aa00)
	noobaa-operator/pkg/system/reconciler.go:422 +0x33b
github.com/noobaa/noobaa-operator/v5/pkg/controller/noobaa.Add.func1({0xc001355cf8?, 0x40dd8a?}, {{{0xc0015f540a?, 0x30?}, {0xc0015f5436?, 0x25507e0?}}})
	noobaa-operator/pkg/controller/noobaa/noobaa_controller.go:53 +0xe5
sigs.k8s.io/controller-runtime/pkg/reconcile.Func.Reconcile(0x7f42a099ba68?, {0x2d1ac00?, 0xc001210a20?}, {{{0xc0015f540a?, 0x0?}, {0xc0015f5436?, 0x40e167?}}})
	noobaa-operator/vendor/sigs.k8s.io/controller-runtime/pkg/reconcile/reconcile.go:103 +0x43
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x2d1ac00?, {0x2d1ac00?, 0xc001210a20?}, {{{0xc0015f540a?, 0x223c760?}, {0xc0015f5436?, 0x2d03fb8?}}})
	noobaa-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:118 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0004d35e0, {0x2d1ab58, 0xc0004ce6e0}, {0x2478bc0?, 0xc001212cc0?})
	noobaa-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:314 +0x377
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0004d35e0, {0x2d1ab58, 0xc0004ce6e0})
	noobaa-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:265 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	noobaa-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:226 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	noobaa-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:222 +0x587
```

--- a/pkg/system/phase4_configuring.go
+++ b/pkg/system/phase4_configuring.go
@@ -1007,6 +1007,10 @@ func (r *Reconciler) prepareGCPBackingStore() error {
 		return err
 	}
 	projectID := authJSON.ProjectID
+	if r.GCPBucketCreds.StringData == nil {
+		r.Logger.Infof("Secret %q does not contain a map of StringData yet. retry on next reconcile...", secretName)
+		return fmt.Errorf("cloud credentials secret %q is not ready yet (does not contain a map of StringData yet)", secretName)
+	}
 	r.GCPBucketCreds.StringData["GoogleServiceAccountPrivateKeyJson"] = cloudCredsSecret.StringData["service_account.json"]
 	ctx := context.Background()
 	gcpclient, err := storage.NewClient(ctx, option.WithCredentialsJSON([]byte(cloudCredsSecret.StringData["service_account.json"])))

